virtualmachine/vsphere/vm.go

```diff
@@ -1393,3 +1428,62 @@ func CreateTemplate(vm *VM) error {
 	err = uploadTemplate(vm, dcMo, vm.datastore)
 	return err
 }
+
+// GetDatastores : Returns the datastores in a host/cluster in a cluster
+func GetTemplateList(vm *VM) ([]map[string]string, error) {
+	vmMoList, err := getVirtualMachines(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	vmList := make([]map[string]string, 0)
+	for _, vmo := range vmMoList {
+		// Filter out the templates
+		if vmo.Config.Template {
+			vmList = append(vmList, map[string]string{
+				"name": vmo.Name,
+				"id":   vmo.Self.Value,
+			})
+		}
+	}
+	return vmList, nil
+}
+
+// GetVirtualMachines : Return the virual machines in a cluster
+func getVirtualMachines(vm *VM) ([]mo.VirtualMachine, error) {
```

Comment:
I think exposing types of a library used internally as part of a public API isn't a good approach. If we have to expose the API, an appropriate wrapper should be written. It is not pertinent for the feature at hand though.